python/mxnet/gluon/probability/distributions/utils.py

```diff
@@ -48,7 +51,10 @@ def compute(value):
         """Return digamma(value)
         """
         if isinstance(value, Number):
-            return sc.digamma(value, dtype='float32')
+            if sc is not None:
+                return sc.digamma(value, dtype='float32')
```

Comment:
why the fixed dtype?

Reply:
That's a question for @xidulu

Reply:
Since _value_  is a a python scalar, the default output’s dtype would be float64, which is inconsistent DeepNumpy’s behavior.

Reply:
@xidulu I see. That makes sense. We do have a flag for determining whether the output type should follow numpy convention or not. You can find it here:
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/util.py#L1076-L1092

It would be great if that convention is observed.

Reply:
Ur right, this decorator should be taking into account. 
I will try to figure out a way to let these utilities be aware of the type flag.